src/pudl/transform/phmsagas.py

```diff
@@ -219,10 +318,31 @@ def _check(df):
         ):
             error = "Percentage of rows with negative percent_unaccounted_for_gas values: {negative_percentage:.2f}"
             logger.info(error)
+            return AssetCheckResult(passed=False, metadata={"errors": error})
+
+        return AssetCheckResult(passed=True)
 
-        if error:
+    @asset_check(asset=spec.asset, blocking=True)
+    def _check_pk_deduplication(df):
+        """Check if the size of filtered non-unique rows exceeds the threshold."""
+        # Identify non-unique groups
+        non_unique_groups = df.groupby(["operator_id_phmsa", "report_number"]).filter(
+            lambda group: len(group) > 1
+        )
+
+        # Apply the filters to non-unique groups
+        filtered_non_unique_rows = non_unique_groups.groupby(
+            ["operator_id_phmsa", "report_number"], group_keys=False
+        ).apply(combined_filter)
+
+        if len(filtered_non_unique_rows) > spec.pk_deduplication_theshold:
+            error = (
+                f"Number of filtered non-unique rows ({len(filtered_non_unique_rows)})\n"
+                f"Exceeds the threshold of {spec.pk_deduplication_theshold}"
+            )
+            logger.info(error)
             return AssetCheckResult(passed=False, metadata={"errors": error})
 
         return AssetCheckResult(passed=True)
 
-    return _check
+    return [_check_percent_unaccounted_for_gas, _check_pk_deduplication]
```

Comment:
Right now these asset checks aren't loaded into dagster, at least partially because the function to produce them isn't called anywhere but possibly also because you're returning a list a checks here rather than just one check. If we aren't parameterizing the checks (e.g., specifying expected values by year) then looking at the way in which multiple checks are defined in `eia860.py` would be a better analogy than the FERC 714 model.